pontoon/translate/views.py

```diff
@@ -21,70 +16,14 @@
 )
 
 
-def static_serve_dev(request, path):
-    """Proxy missing static files to the frontend server.
-
-    This view replaces django's static files serve view. When a file is
-    missing from django's paths, then we make a proxy request to the
-    frontend server to see if it's a front-end file.
-
-    Note that to enable this view, you need to run your django with the
-    nostatic option, like this::
-
-    $ ./manage.py runserver --nostatic
-
-    """
-    try:
-        # First try to load the file with django's regular serve view.
-        return serve(request, path, insecure=True)
-    except Http404:
-        # If the file couldn't be found in django's static files, then we
-        # try to proxy it to the frontend server.
-        return catchall_dev(request)
-
-
 @csrf_exempt
 def catchall_dev(request, context=None):
-    """Proxy HTTP requests to the frontend dev server in development.
-
-    The implementation is very basic e.g. it doesn't handle HTTP headers.
-
-    """
-    # URL to the development frontend server, used to redirect front-end requests.
-    upstream_url = settings.FRONTEND_URL + request.path
-
-    # Redirect websocket requests directly to the frontend server.
-    if request.META.get("HTTP_UPGRADE", "").lower() == "websocket":
-        return http.HttpResponseRedirect(upstream_url)
-
-    method = request.META["REQUEST_METHOD"].lower()
-    response = getattr(requests, method)(upstream_url, stream=True)
-    content_type = response.headers.get("Content-Type")
-
-    if content_type == "text/html; charset=UTF-8":
-        return http.HttpResponse(
-            content=(
-                engines["jinja2"]
-                .from_string(response.text)
-                .render(request=request, context=context)
-            ),
-            status=response.status_code,
-            reason=response.reason,
-        )
-
-    return http.StreamingHttpResponse(
-        streaming_content=response.iter_content(2 ** 12),
-        content_type=content_type,
-        status=response.status_code,
-        reason=response.reason,
-    )
+    return render(request, "frontend.html", context=context, using="jinja2")
 
 
-# For production, we've added the front-end static files to our list of
-# static folders. We can thus simply return a template view of index.html.
 @ensure_csrf_cookie
 def catchall_prod(request, context=None):
-    return render(request, "index.html", context=context, using="jinja2")
+    return render(request, "frontend.html", context=context, using="jinja2")
```

Comment:
Seems we only need one of `catchall_*` now?

Reply:
The CSRF decorators are different on `catchall_dev` and `catchall_prod`. As I wasn't completely certain about what they did, I left them here as separate.